modules/core/src/lib/attribute.js

```diff
@@ -176,47 +187,57 @@ export default class Attribute extends BaseAttribute {
         this.buffer.reallocate(this.value.byteLength);
       }
 
-      state.needsUpdate = true;
+      if (state.updateRanges !== range.FULL) {
+        this.value.set(oldValue);
+        // TODO - copy on the GPU
+        this.buffer.subData(oldValue);
```

Comment:
Maybe some comment here. I would naively expect we would need to specify an offset if range is not `FULL`?